src/main/java/com/leff/midi/MidiFile.java

```diff
@@ -37,52 +34,25 @@
     private int mType;
     private int mTrackCount;
     private int mResolution;
+    private List<MidiTrack> mTracks = new ArrayList<>();
 
-    private ArrayList<MidiTrack> mTracks;
-
-    public MidiFile() {
-        this(DEFAULT_RESOLUTION);
-    }
-
-    public MidiFile(int resolution) {
-        this(resolution, new ArrayList<MidiTrack>());
-    }
-
-    public MidiFile(int resolution, ArrayList<MidiTrack> tracks) {
-        mResolution = resolution >= 0 ? resolution : DEFAULT_RESOLUTION;
-
-        mTracks = tracks != null ? tracks : new ArrayList<MidiTrack>();
-        mTrackCount = tracks.size();
-        mType = mTrackCount > 1 ? 1 : 0;
-    }
-
-    public MidiFile(File fileIn) throws FileNotFoundException, IOException {

```

Comment:
Remove all file-related concepts from this file.  It only works with `byte[]` or `InputStream`